Popover component IE compatibility

rmc00 opened the pull request:
Each popover component was sending Events to close other popovers when opening. Since the `new Event('hide')` syntax is not supported in IE11, this change will increase cross-browser compatibility. Without this, you can get errors if a popover is expanded when another popover tries to expand. This would fix Issue #846 .

tb commented on src/components/popover/popover.ts:
```diff
@@ -66,7 +66,9 @@ export class MdlPopoverComponent implements AfterViewInit {
       for(let i=0; i < nodeList.length;++i) {
-        nodeList[i].dispatchEvent(new Event('hide'));
+          let event = document.createEvent('CustomEvent');
+          event.initCustomEvent('hide', false, false, {});
```
What about https://msdn.microsoft.com/en-us/library/ff975299(v=vs.85).aspx

> Important

  As of Microsoft Edge, the createEvent()/initEvent() constructor pattern for synthetic events is deprecated. See the Synthetic Events page for more information.

and

https://developer.mozilla.org/en/docs/Web/API/Event/initEvent

> Do not use this method anymore as it is deprecated.
> 
> Instead use specific event constructors, like Event(). The page on Creating and triggering events gives more information about the way to use these.
>

rmc00 commented:
@tb Thanks for your feedback. You're correct about how I was creating the CustomEvent before, so I've gone back to the `new CustomEvent('hide')` syntax, which should hopefully resolve your concern, and I added a small polyfill for that constructor based on this one from [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill).

tb commented on src/e2e-app/polyfills.ts:
```diff
@@ -9,3 +9,16 @@ if (process.env['ENV'] === 'production') {
   require('zone.js/dist/long-stack-trace-zone');
 }
+
+(function () {
```
This code wont be distributed with popover package. @mseemann where should it be put? Maybe in `/src/components/popover/custom-event-polyfil.ts`? There is also npm package with this polyfil... https://www.npmjs.com/package/custom-event-polyfill that could be added as dependency of the popover package.

mseemann replied:
in this case i would suggest going the same way how all other polyfills that are needed for angular are used. (for example angular cli generates a polyfills.ts (https://github.com/mseemann/angular-mdl-a4-angular-cli/blob/master/src/polyfills.ts) 

also there seems to be only one browser that did not fully support CustomEvents http://caniuse.com/#feat=customevent

->
- no polyfill code in the package
- a hint in the documentation how to add the polyfill

@tb did you agree?

rmc00 commented:
@mseemann @tb Thanks for the feedback on handling polyfills. I went ahead and removed the polyfill from the E2E app. I can add the polyfill into a `polyfills.ts` file in the popover package if you'd like. At least this way, the IE11 bug can be resolved by adding the CustomEvent polyfill into applications, which I don't think could be done with regular events before.

rmc00 commented:
@mseemann @tb Is it possible to merge this PR? I'd love to move forward with this if possible. I think I've addressed all feedback -- please let me know if that is not the case.

tb commented:
@mseemann I think its fine, it comes down to change `Event` to `CustomEvent`. However we might need to mention including https://www.npmjs.com/package/custom-event-polyfill

mseemann commented:
ok. i have to increase the versions for select and fab-menu too - because of the umd bundle problem.

mseemann commented:
published
